src/openslide-vendor-philips-isyntax.c

```diff
@@ -0,0 +1,397 @@
+    if (str_is_global_cache && *str_is_global_cache == '0') {
+        is_global_cache = false;
+    }
+    if (str_cache_size) {
+        cache_size = atoi(str_cache_size);
+    }
+    /* TODO(avirodov): make debug api.
+    {
+        uint64_t memory_count = 0;
+        isyntax_image_t* wsi = &data->isyntax->images[data->isyntax->wsi_image_index];
+        for (int level_idx = 0; level_idx < wsi->level_count; ++level_idx) {
+            memory_count += wsi->levels[level_idx].tile_count * sizeof(isyntax_tile_t);
+        }
+        memory_count += wsi->codeblock_count * sizeof(isyntax_codeblock_t);
+        memory_count += wsi->data_chunk_count * sizeof(isyntax_data_chunk_t);
+        printf("philips_isyntax_open is_global_cache=%d cache_size=%d sizeof_structs=%'lld\n", (int)is_global_cache, cache_size, memory_count);
+    } */
+    if (is_global_cache) {
+        g_mutex_lock(&static_open_mutex);
+        if (philips_isyntax_global_cache_ptr == NULL) {
+            // Note: this requires that all opened files have the same block size. If that is not true, we
+            // will need to have allocator per size. Alternatively, implement allocator freeing after
+            // all tiles have been freed, and track isyntax_t per tile so we can access allocator.
+            philips_isyntax_global_cache_ptr = malloc(sizeof(*philips_isyntax_global_cache_ptr));
+            ASSERT_OK(libisyntax_cache_create("global_cache_list", cache_size, &philips_isyntax_global_cache_ptr->cache));
+        }
+        data->cache = philips_isyntax_global_cache_ptr;
+        g_mutex_unlock(&static_open_mutex);
+    } else {
+        ASSERT_OK(libisyntax_cache_create("cache_list", cache_size, &data->cache->cache));
+    }
+    // Link the cache (local or global) to the isyntax file.
+    libisyntax_cache_inject(data->cache->cache, data->isyntax);
+
+    LOG_VAR("%d", libisyntax_get_is_mpp_known(data->isyntax));
+    if (libisyntax_get_is_mpp_known(data->isyntax)) {
+        double mpp_x = libisyntax_get_mpp_x(data->isyntax);
+        double mpp_y = libisyntax_get_mpp_y(data->isyntax);
+        LOG_VAR("%f", mpp_x);
+        LOG_VAR("%f", mpp_y);
+        add_float_property(osr, OPENSLIDE_PROPERTY_NAME_MPP_X, mpp_x);
+        add_float_property(osr, OPENSLIDE_PROPERTY_NAME_MPP_Y, mpp_y);
+        const float float_equals_tolerance = 1e-5;
+        if (fabs(mpp_x - mpp_y) < float_equals_tolerance) {
+            // Compute objective power from microns-per-pixel, see e.g. table in "Scan Performance" here:
+            // https://www.microscopesinternational.com/blog/20170928-whichobjective.aspx
+            float objective_power = 10.0f / mpp_x;
+            LOG_VAR("%f", objective_power);
+            add_float_property(osr, OPENSLIDE_PROPERTY_NAME_OBJECTIVE_POWER, objective_power);
+        }
+    }
+
+    // Find wsi image. Extracting other images not supported. Assuming only one wsi.
+    int wsi_image_idx = libisyntax_get_wsi_image_index(data->isyntax);
```

Comment:
Again, some of that is being discussed in https://github.com/amspath/libisyntax/issues/4 "Role of wsi_image_t".

From implementation, since there is only one `isyntax_t.wsi_image_index`, there can be only one pyramidal image exposed by libisyntax.

The API here will change though I expect, maybe let's defer to that. Not very soon though, so I hope not to block on that.